### PR TITLE
Adds README/GODeps for specgen tool.

### PR DESCRIPTION
**Pull request by jterry75**
1. Adds the support for GoDep's for specgen to tie the git hash to the
generated model.
2. Adds information to the readme to help people know how to use the tool.

Resolves: #52

Signed-off-by: Justin Terry <juterry@microsoft.com>

### CODE REVIEW

**Comment by jterry75** [comment]
@jstarks - This completes the work to make the generator and documentation part of Docker.DotNet.

File: tools/specgen/README.md [hunk 1]
```diff
@@ -1,2 +1,101 @@
-# docker-engine-api-cs
-Go code to generate C# classes from docker's engine-api.
+# SpecGen
+
+A tool that reflects the Docker client [engine-api](https://github.com/docker/engine-api) in order to generate C# classes that match its model for [Docker.DotNet Models](https://github.com/ahmetalpbalkan/Docker.DotNet/tree/master/Docker.DotNet/Models).
+
+<hr />
+<h2>How to use:</h2>
```
Comment (ahmetalpbalkan) [inline tools/specgen/README.md]: nit: you can use `##` for h2, and `-----` for hr.

**Comment by jterry75** [comment]
@swernli  - Since John is out.

**Comment by swernli** [comment]
LGTM.

**Pull request merged**